scipy/optimize/_simulated_annealing.py

```diff
@@ -0,0 +1,559 @@
+        `np.random.RandomState` instance is used.
+        Specify `seed` for repeatable minimizations. The random numbers
+        generated with this seed only affect the visiting distribution
+        function and new coordinates generation.
+    no_local_search : boolean, optional
+        If `no_local_search` is set to `True`, a traditional Generalized
+        Simulated Annealing will be performed with no local search
+        strategy applied.
+
+    Returns
+    -------
+    res : OptimizeResult
+        The optimization result represented as a ``OptimizeResult`` object.
+        Important attributes are: ``x`` the solution array, ``fun`` the value
+        of the function at the solution, and ``message`` which describes the
+        cause of the termination.
+        See `OptimizeResult` for a description of other attributes.
+
+    Notes
+    -----
+    SDA is an implementation of the Simulated Dual Annealing. This stochastic
+    approach [3]_ implements the generalization of CSA (Classical Simulated
+    Annealing) and FSA (Fast Simulated Annealing) [1]_ [2]_ coupled to a
+    strategy for applying a local search on accepted locations [4]_. A first
+    implementation was done in C++ for the R language [5]_ and has been
+    benchmarked [6]_. Extensive benchmarks for the Python version have also
+    been performed against several global optimization methods [7]_. SDA
+    introduces an advanced dual annealing method to refine the solution found
+    by the generalized annealing process. This algorithm uses a distorted
+    Cauchy-Lorentz visiting distribution, with its shape controlled by the
+    parameter :math:`q_{v}`
+
+    .. math::
+
+        g_{q_{v}}(\\Delta x(t)) \\propto \\frac{ \\
+        \\left[T_{q_{v}}(t) \\right]^{-\\frac{D}{3-q_{v}}}}{ \\
+        \\left[{1+(q_{v}-1)\\frac{(\Delta x(t))^{2}} { \\
+        \\left[T_{q_{v}}(t)\\right]^{\\frac{2}{3-q_{v}}}}}\\right]^{ \\
+        \\frac{1}{q_{v}-1}+\\frac{D-1}{2}}}
+
+    Where :math:`t` is the artificial time. This visiting distribution is used
+    to generate a trial jump distance :math:`\Delta x(t)` of variable
+    :math:`x(t)` under artificial temperature :math:`T_{q_{v}}(t)`.
+
+    From the starting point, after calling the visiting distribution
+    function, the acceptance probability is computed as follows:
+
+    .. math::
+
+        p_{q_{a}} = \min{\{1,\\left[1-(1-q_{a}) \\beta \\Delta E \\right]^{ \\
+        \\frac{1}{1-q_{a}}}\\}}
+
+    Where :math:`q_{a}` is a acceptance parameter. For :math:`q_{a}<1`, zero
+    acceptance probability is assigned to the cases where
+
+    .. math::
+
+        [1-(1-q_{a}) \\beta \\Delta E] < 0
+
+    The artificial temperature :math:`T_{q_{v}}(t)` is decreased according to
+
+    .. math::
+
+        T_{q_{v}}(t) = T_{q_{v}}(1) \\frac{2^{q_{v}-1}-1}{\\left( \\
+        1 + t\\right)^{q_{v}-1}-1}
+
+    Where :math:`q_{v}` is the visiting parameter.
+
+    .. versionadded:: 1.1.0
+
+    References
```

Comment:
Please put all the references in a standard format.

Reply:
Reformatted.